### PR TITLE
feat: add metric for batch process entries

### PR DESCRIPTION
**Pull request by gy09535**
### What this PR does / why we need it:

fix: https://github.com/apache/apisix/issues/2693

add metric for batch process, such as:  sls logger, tcp logger ,udp logger , http logger, sys logger. 
with this feature, users can use the metrics find how many entries remaining in batch process.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by membphis** [comment]
@gy09535 need a clear name about what you are doing.

**Comment by gy09535** [comment]
> @gy09535 need a clear name about what you are doing.

got it.

File: apisix/plugins/http-logger.lua [hunk 1]
```diff
@@ -277,7 +277,8 @@ function _M.log(conf, ctx)
         max_retry_count = conf.max_retry_count,
         buffer_duration = conf.buffer_duration,
         inactive_timeout = conf.inactive_timeout,
-        route_id = ctx.route_id
+        route_id = ctx.route_id,
+        server_addr = ctx.var["server_addr"]
```
Comment (membphis) [inline apisix/plugins/http-logger.lua:281]: code style, we should always add a `,`  here.

example:

```lua
        route_id = ctx.route_id,
        server_addr = ctx.var["server_addr"],

Comment (membphis) [reply apisix/plugins/http-logger.lua:281]: and please fix similar points

File: apisix/plugins/http-logger.lua [hunk 2]
```diff
@@ -277,6 +277,8 @@ function _M.log(conf, ctx)
         max_retry_count = conf.max_retry_count,
         buffer_duration = conf.buffer_duration,
         inactive_timeout = conf.inactive_timeout,
+        route_id = ctx.route_id,
```
Comment (membphis) [inline apisix/plugins/http-logger.lua:280]: for `route_id`, we should ways use `ctx.var.route_id`

it is the recommended way. and please fix similar points.

**Comment by gy09535** [comment]
@membphis  please  help me remove this PR, I will push  another one , I have put some sensitive info in this PR. Thank you very much.

**Comment by membphis** [comment]
asdf

**Pull request closed without merging**